guides/source/action_text_overview.md

```diff
@@ -264,10 +264,13 @@ not being installed.
 
 | Event name | Event target | Event data (`event.detail`) | Description |
 | --- | --- | --- | --- |
-| `direct-upload:start` | `<input>` | `{id, file}` | A direct upload is starting. |
```

Comment:
This PR combines three types of changes:

1. it changes incorrect documentation to reflect the actual pre-existing behavior (for example, changes `<input>` to `<trix-editor>` and adds `attachment` to the `event.details`)
2. it adds missing behavior to reflect the documentation (for example, adds `id` and `file` to the dispatched events)
3. adds new behavior to achieve parity with Active Storage (for example, adds `direct-upload:initialize` and the `direct-upload:before-` events)

I wasn't sure how to best resolve all 3. At the moment, doing all 3 at once feels like it prioritizes minimizing code review efforts. All three are also supported by the System Test coverage, which does not already exist.

If we think the changes are too ambitious, I'm happy to split the PR into one that handles 1. and 2. together, and then a separate one that handles 3. on its own.

Reply:
Upon further consideration, I think 1. and 2. could be backported to older releases (maybe just to 8.0.0?), whereas 3. is new behavior.

Whether or not 3. is a "fix" might be open to interpretation, so I defer to Core on how to proceed.